src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php

```diff
@@ -337,4 +337,25 @@ protected function compileDisabled($condition)
     {
         return "<?php if{$condition}: echo 'disabled'; endif; ?>";
     }
+
+    /**
+     * Compile a route-is into valid PHP.
+     *
+     * @param  string|array  $route
+     * @return string
+     */
+    protected function compileRouteIs($route)
+    {
+        return "<?php if(request()->routeIs($route)): ?>";
+    }
```

Comment:
The `$route` parameter includes parenthesis from the Blade directive so `@routeIs('home')` would compile to `"<?php if(request()->routeIs(('home')): ?>";`.


The method supports variadic arguments:
https://github.com/laravel/framework/blob/503f6e280c2978ad8e57f995bec07515eeef19ae/src/Illuminate/Http/Request.php#L223-L226

So the raw Blade directive string argument can be retained to support multiple arguments. e.g., `@routeIs('home', 'about')`:


```suggestion
    protected function compileRouteIs($patterns)
    {
        return "<?php if (request()->routeIs{$patterns}): ?>";
    }
```

Tests should also be added in https://github.com/laravel/framework/tree/9.x/tests/View/Blade to show the directive keeps compiling to valid PHP syntax.

Reply:
Thank You I will write test tomorrow and make necessary changes